From af2bf246dbd2e81ad621efe8c47c3203311349c6 Mon Sep 17 00:00:00 2001
From: Kalle Lampila <kallex.lampila@intel.com>
Date: Tue, 30 Apr 2013 15:27:19 +0300
Subject: [PATCH] [PORT FROM MAIN] Camera service: fix deadlock

BZ: 104720

Release lock before return from CameraService::dump().

Change-Id: Ia20b0921fc675a284f3b7f8d26e0e3596b5a8774
Orig-Change-Id: I89d75f13bf82490929eafa1da9e88e7023abfb49
Category: aosp improvement
Domain: camera
Origin: internal
Upstream-Candidate: yes
Signed-off-by: Kalle Lampila <kallex.lampila@intel.com>
---
 services/camera/libcameraservice/CameraService.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/services/camera/libcameraservice/CameraService.cpp b/services/camera/libcameraservice/CameraService.cpp
index 594e2f7..5c1e837 100644
--- a/services/camera/libcameraservice/CameraService.cpp
+++ b/services/camera/libcameraservice/CameraService.cpp
@@ -1193,6 +1194,7 @@ status_t CameraService::dump(int fd, const Vector<String16>& args) {
         if (!mModule) {
             result = String8::format("No camera module available!\n");
             write(fd, result.string(), result.size());
+            if (locked) mServiceLock.unlock();
             return NO_ERROR;
         }
 
-- 
2.5.0

